src/scripts/Changelog.ts

```diff
@@ -9,15 +24,20 @@ const changelogType = {
 };
 
 class Changelog {
-    public type: { display: string; label: string };
-    public description: string;
-
-    constructor(type = { display: 'default', label: '-' }, description: string) {
-        this.type = type;
-        this.description = description;
-        if (type == changelogType.UPDATE) {
-            this.description = `<code>${this.description}</code>`;
-        }
+    constructor(
+        public type: ChangelogConfig = changelogType.DEFAULT,
+        public description: string
+    ) {}
+}
+
+class ChangelogUpdate extends Changelog {
+    constructor(
+        version: string,
+        date: Date
+    ) {
+        super(changelogType.UPDATE, version);
+        const dateFormat: Intl.DateTimeFormatOptions = {year: 'numeric', month: 'long', day: 'numeric'};
+        this.description = `<code>${this.description}  -  ${date.toLocaleDateString(undefined, dateFormat)}</code>`;
```

Comment:
Minor, but it might be cleaner to push the description through the constructor, instead of initially setting it to `version`
```suggestion
        const dateFormat: Intl.DateTimeFormatOptions = {year: 'numeric', month: 'long', day: 'numeric'};
        const description = `<code>${version}  -  ${date.toLocaleDateString(undefined, dateFormat)}</code>`;
        super(changelogType.UPDATE, description);
```

Reply:
Definitely better. I thought I had tried it this way first, but had typescript shout at me for daring to do anything before calling super, but it doesn't happen when I try it now...

Reply:
It will scream if anything touches `this` before `super`, but as long as we're only using `const` with no helper methods, it'll usually be okay